hvac/adapters.py

```diff
@@ -362,6 +362,10 @@ def request(self, *args, **kwargs):
         :rtype: dict | requests.Response
         """
         response = super(JSONAdapter, self).request(*args, **kwargs)
+        # Do not attempt to jsonify stream requests, doing so will empty the
+        # socket and the caller has explicitly opted out
+        if kwargs.get('stream', False):
+            return response
         if response.status_code == 200:
```

Comment:
FWIW, I'm of the opinion this check should instead only attempt to JSON() the response iif the content-type is `application/json`.  It seemed out of scope for this particular change, so I didn't modify it.

Something like:
```
if response.ok  and 'application/json' in response.headers.get('content-type', ''):
    ...
```

I do see some API responses are documented to 204 status and an empty body though they clearly return data, it's unclear to me what the intent is. So I'm leaving this alone.

Reply:
https://www.vaultproject.io/api#http-status-codes says the following:

> Applications should be prepared to accept both 200 and 204 as success. 204 is simply an indication that there is no response body to parse, but API endpoints that indicate that they return a 204 may return a 200 if warnings are generated during the operation.

So it seems that 204 should be a solid indication there is no body, while if there is a body (due to warnings or other circumstances) then the code should be 200.

If you've seen instances of that not being the case, it would be good to know.

Reply:
Ooh nice @briantist, I was unaware of that!